apply-dictionary-renderer-to-feature-layer/src/main/java/com/esri/arcgismaps/sample/applydictionaryrenderertofeaturelayer/MainActivity.kt

```diff
@@ -0,0 +1,117 @@
+import java.io.File
+
+class MainActivity : AppCompatActivity() {
+
+    private val TAG = MainActivity::class.java.simpleName
+
+    // set up data binding for the activity
+    private val activityMainBinding: ActivityMainBinding by lazy {
+        DataBindingUtil.setContentView(this, R.layout.activity_main)
+    }
+
+    private val mapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    private val provisionPath: String by lazy {
+        getExternalFilesDir(null)?.path.toString() + File.separator + getString(R.string.app_name)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
+        lifecycle.addObserver(mapView)
+
+        // create and add a map with a navigation night basemap style
+        val map = ArcGISMap(BasemapStyle.ArcGISTopographic)
+        mapView.map = map
+
+        // locate the .stylx file in the device
+        val styleFile = File(provisionPath, getString(R.string.mil2525d_stylx))
+        // instantiate the dictionarySymbolStyle using the file path
+        val dictionarySymbolStyle = DictionarySymbolStyle.createFromFile(styleFile.absolutePath)
+
+        // locate the .geodatabase file in the device
+        val geodatabaseFile = File(provisionPath, getString(R.string.militaryoverlay_geodatabase))
+        // instantiate the geodatabase with the file path
+        val geodatabase = Geodatabase(geodatabaseFile.path)
+
+        lifecycleScope.launch {
+            // load the dictionary symbol style
+            dictionarySymbolStyle.load().getOrElse {
+                return@launch showError("Error loading DictionarySymbolStyle: ${it.message}")
+            }
+
+            // load the geodatabase
+            geodatabase.load().getOrElse {
+                showError("Error loading Geodatabase: ${it.message}")
+            }
+
+            geodatabase.featureTables.forEach { geodatabaseFeatureTable ->
+                // load each geodatabaseFeatureTable and create featureLayer from it
+                geodatabaseFeatureTable.load().getOrElse {
+                    return@launch showError("Error loading GeodatabaseFeatureTable: ${it.message}")
+                }
+                val featureLayer = FeatureLayer(geodatabaseFeatureTable)
+                featureLayer.load().getOrElse {
+                    return@launch showError("Error loading FeatureLayer: ${it.message}")
+                }
+                // add featureLayer to the map's operational layer
+                mapView.map?.operationalLayers?.add(featureLayer)
+
+                // create dictionaryRenderer using the dictionarySymbolStyle and apply it to the featureLayer's renderer
+                val dictionaryRenderer = DictionaryRenderer(dictionarySymbolStyle)
```

Comment:
Would it work to create a single DictionaryRenderer and apply it to every feature layer? That's a bit of a nitpick, and I'm not sure the optimization wouldn't just make it harder to read the sample, so I would leave it up to you.